src/sentry/search/base.py

```diff
@@ -14,36 +14,13 @@
 
 
 class SearchBackend(Service):
-    __all__ = frozenset([
-        'query',
-    ])
+    __read_methods__ = frozenset(['query'])
+    __write_methods__ = frozenset()
+    __all__ = __read_methods__ | __write_methods__
 
     def __init__(self, **options):
         pass
 
-    def query(
-        self,
-        project,
-        query=None,
-        status=None,
-        tags=None,
-        bookmarked_by=None,
-        assigned_to=None,
-        first_release=None,
-        sort_by='date',
-        age_from=None,
-        age_to=None,
-        unassigned=None,
-        date_from=None,
-        date_to=None,
-        cursor=None,
-        limit=100
-    ):
-        """
-        The return value should be a CursorResult.
-
-        The limit here is a soft input limit, which gets trimmed by the Cursor.
-        This means the backend should query limit + 2 and return that within the
-        CursorResult.
-        """
+    def query(self, project, tags=None, environment=None, sort_by='date', limit=100,
+              cursor=None, count_hits=False, paginator_options=None, **parameters):
```

Comment:
This is required since the delegator ensures that the provided arguments are valid with respect to the base backend before passing the arguments onto the secondaries. This is basically just a copy/paste from [the Django backend `query` definition](https://github.com/getsentry/sentry/blob/5870b8aae55a8f969fdb2454b8dea7ecdacbeb4a/src/sentry/search/django/backend.py#L207-L208).

It's not great (this basically provides no enforcement or hinting about the set of parameters that the backends should implement) but seems like the least painful way to keep this in sync for now and not really any worse than it was previously.